core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4664,8 +4664,8 @@ private void analyzeRowFilter(String currentIdentity, Table table, QualifiedObje
             ExpressionAnalysis expressionAnalysis;
             try {
                 Identity filterIdentity = filter.getSecurityIdentity()
-                        .map(filterUser -> Identity.forUser(filterUser)
-                                .withGroups(groupProvider.getGroups(filterUser))
+                        .map(identity -> Identity.from(identity)
+                                .withGroups(groupProvider.getGroups(identity.getUser()))
```

Comment:
I guess if you take the identity from there, you should just use it without adding groups to it.

Reply:
Like I said in https://github.com/trinodb/trino/pull/16729#discussion_r1148925431, groups are managed by Trino, not by access controls, so we still have to get them from the group provider. Another question is what to do if access control *does* return groups in the `Identity`.